eclair-core/src/main/scala/fr/acinq/eclair/router/Validation.scala

```diff
@@ -156,6 +160,38 @@ object Validation {
     }
   }
 
+  private def splicePublicChannel(d: Data, nodeParams: NodeParams, watcher: typed.ActorRef[ZmqWatcher.Command], ann: ChannelAnnouncement, spliceTxId: TxId, capacity: Satoshi, parentChannel: PublicChannel)(implicit ctx: ActorContext, log: DiagnosticLoggingAdapter): Data = {
+    implicit val sender: ActorRef = ctx.self // necessary to preserve origin when sending messages to other actors
+    val fundingOutputIndex = outputIndex(ann.shortChannelId)
+    watcher ! WatchExternalChannelSpent(ctx.self, spliceTxId, fundingOutputIndex, ann.shortChannelId)
+    ctx.system.eventStream.publish(ChannelsDiscovered(SingleChannelDiscovered(ann, capacity, None, None) :: Nil))
+    nodeParams.db.network.addChannel(ann, spliceTxId, capacity)
+    nodeParams.db.network.removeChannel(parentChannel.shortChannelId)
+    val pubChan = PublicChannel(
+      ann = ann,
+      fundingTxId = spliceTxId,
+      capacity = capacity,
+      update_1_opt = None,
+      update_2_opt = None,
```

Comment:
Good point; I found two places here where I set `update_1_opt` and `update_2_opt` to None. 
1) for the `ChannelUpdate`s of the new `SingleChannelDiscovered` we publish for the `FrontRouter` 
 -  we don't want the `FrontRouter` to rebroadcast modified `ChannelUpdates` with signatures that are invalid
2) when creating the new `PublicChannel` for our graph we can reuse the previous values for `ChannelUpdate`s, but we should update the timestamps so the channel won't be pruned. 

Is there some guarantee that we'll always get a new `ChannelUpdate` for these newly spliced channels?

Reply:
It also seems like if we add a new channel to the `FrontRouter` with `ChannelsDiscovered` we should also call `ChannelLost` with the previous scid, otherwise they'll just accumulate in the `FrontRouter`.

Reply:
> Shouldn't we keep the channel updates we had for the pre-splice channel? Otherwise the channel will be unusable until we receive the new channel updates. It's slightly weird because we will temporarily have channel updates that use the previous `scid`, but I don't think this has any negative side effect?

Fixed in [43a486e](https://github.com/ACINQ/eclair/pull/2936/commits/43a486e7e121643d1eafe3deffee1a9d44ee0ed3) . 

I copy the `update_1_opt` and `update_2_opt` from the pre-splice `PublicChannel`, but update their `shortChannelId` to match the splice announcements `shortChannelId` so we can derive the new edge in the graph from this `ChannelUpdate`. 

I also reset the `timestamp` to the current time to prevent the spliced channel from being pruned prematurely if the post-splice `ChannelUpdate` is not received due to unreliable gossip.